src/Microsoft.AspNet.Hosting/Internal/HostingApplication.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Hosting.Server;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Hosting.Internal
+{
+    public class HostingApplication : IHttpApplication
+    {
+        private readonly IServiceProvider _applicationServices;
+        private readonly RequestDelegate _application;
+        private readonly ILogger _logger;
+        private readonly DiagnosticSource _diagnosticSource;
+        private readonly IHttpContextFactory _httpContextFactory;
+
+        public HostingApplication(
+            IServiceProvider applicationServices,
+            RequestDelegate application)
+        {
+            _applicationServices = applicationServices;
+            _application = application;
+            _logger = _applicationServices.GetRequiredService<ILogger<HostingEngine>>();
+            _diagnosticSource = _applicationServices.GetRequiredService<DiagnosticSource>();
+            _httpContextFactory = _applicationServices.GetRequiredService<IHttpContextFactory>();
+        }
+
+        public HttpContext CreateHttpContext(IFeatureCollection contextFeatures)
+        {
+            return _httpContextFactory.Create(contextFeatures);
```

Comment:
I dunno - if the server calls LogWarning(ex) it'll work perfectly well. Is it that we want to have the same category & eventid on unhandled exception?